### PR TITLE
MAINT: Add a common subclass to all the masked ufunc wrappers

### PR DESCRIPTION
**Pull request by eric-wieser**
Could probably go further than this, but it's at least a start

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -1136,12 +1139,10 @@ def __init__(self, dbfunc, domain, fillx=0, filly=0):
         """abfunc(fillx, filly) must be defined.
            abfunc(x, filly) = x for all x to enable reduce.
         """
-        self.f = dbfunc
+        super(_DomainedBinaryOperation, self).__init__(dbfunc)
         self.domain = domain
         self.fillx = fillx
         self.filly = filly
-        self.__doc__ = getattr(dbfunc, "__doc__", str(dbfunc))
-        self.__name__ = getattr(dbfunc, "__name__", str(dbfunc))
```
Comment (eric-wieser) [inline numpy/ma/core.py]: I don't know what this was trying to catch, but it works fine with a direct lookup.

File: numpy/ma/core.py [hunk 2]
```diff
@@ -880,7 +880,17 @@ def __call__(self, x):
...
 
-class _MaskedUnaryOperation(object):
+class _MaskedUFunc(object):
+    def __init__(self, ufunc):
+        self.f = ufunc
+        self.__doc__ = ufunc.__doc__
+        self.__name__ = ufunc.__name__
+
+    def __str__(self):
+        return "Masked version of {}".format(self.f)
```
Comment (eric-wieser) [inline numpy/ma/core.py]: No idea why this isn't `__repr__`, but this at least maintains the old behaviour

File: numpy/ma/core.py (hunk 1 above)
Comment (charris) [inline numpy/ma/core.py]: Guess we will see if it matter. That line is pretty darn old...

**Comment by charris** [comment]
Thanks Eric.  I was a bit on the fence about whether this needed a release note, but it seems more of a straight refactor than any sort of enhancement.

**Pull request merged**